tests/Unit/ActionTesting.hpp

```diff
@@ -99,17 +106,20 @@ class MockLocalAlgorithm {
   // The next action we should execute.
   size_t algorithm_step_ = 0;
   bool performing_action_ = false;
+
+  typename Component::index array_index_{};
+  Parallel::ConstGlobalCache<typename Component::metavariables>*
+      const_global_cache_{nullptr};
+  tuples::TaggedTupleTypelist<
+      Parallel::get_inbox_tags<typename Component::action_list>>* inboxes_{
+      nullptr};
 };
 
 template <typename Component>
 template <size_t... Is>
 void MockLocalAlgorithm<Component>::next_action(
-    std::index_sequence<Is...> /*meta*/,
-    const typename Component::index& array_index,
-    const gsl::not_null<tuples::TaggedTupleTypelist<inbox_tags_list>*> inboxes,
-    const gsl::not_null<Parallel::ConstGlobalCache<metavariables>*>
-        const_global_cache_ptr) noexcept {
-  auto& const_global_cache = *const_global_cache_ptr;
+    std::index_sequence<Is...> /*meta*/) noexcept {
+  auto& const_global_cache = *const_global_cache_;
   if (performing_action_) {
     ERROR(
```

Comment:
(General question) Do you expect this to be a common mistake people writing actions make, or will it be hard to trigger?

Reply:
I don't think it should ever be triggered. The AlgorithmImpl can be cleaned up a bit, so I'll add an issue describing this and the other things I've noticed. I'll add a separate PR to handle that.